src/pages/TermsOfService.test.ts

```diff
@@ -0,0 +1,228 @@
+    },
+  });
+  const getFenceStatus = jest.fn();
+  const getNihStatus = jest.fn();
+
+  type AjaxContract = ReturnType<typeof Ajax>;
+  type UserContract = ReturnType<typeof User>;
+  type MetricsContract = ReturnType<typeof Metrics>;
+  type GroupsContract = ReturnType<typeof Groups>;
+
+  asMockedFn(Ajax).mockImplementation(
+    () =>
+      ({
+        Metrics: {
+          captureEvent: jest.fn(),
+        } as Partial<MetricsContract>,
+        User: {
+          profile: {
+            get: jest.fn().mockResolvedValue({ keyValuePairs: [] }),
+            set: jest.fn().mockResolvedValue({ keyValuePairs: [] }),
+            setPreferences: jest.fn().mockResolvedValue({}),
+            preferLegacyFirecloud: jest.fn().mockResolvedValue({}),
+          },
+          getTos,
+          getTermsOfServiceComplianceStatus,
+          getStatus,
+          acceptTos,
+          rejectTos,
+          getFenceStatus,
+          getNihStatus,
+        } as Partial<UserContract>,
+        Groups: {
+          list: jest.fn(),
+        } as Partial<GroupsContract>,
+      } as Partial<AjaxContract> as AjaxContract)
+  );
+
+  const signInStatus = 'signedIn';
+  authStore.update((state) => ({ ...state, termsOfService, signInStatus }));
+  return {
+    getTosFn: getTos,
+    getStatusFn: getStatus,
+    getTermsOfServiceComplianceStatusFn: getTermsOfServiceComplianceStatus,
+    acceptTosFn: acceptTos,
+    rejectTosFn: rejectTos,
+  };
+};
+
+describe('TermsOfService', () => {
+  it('fetches the Terms of Service text from Sam', async () => {
+    // Arrange
+    const termsOfService = {
+      userHasAcceptedLatestTos: true,
+      permitsSystemUsage: true,
+    };
+
+    const { getTosFn } = setupMockAjax(termsOfService);
+
+    // Act
+    await act(async () => {
+      render(h(TermsOfServicePage));
+    });
+
+    // Assert
+    expect(getTosFn).toHaveBeenCalled();
+
+    screen.getByText('some text');
+  });
+
+  it('shows "Continue under grace period" when the user has not accepted the latest ToS but is still allowed to use Terra', async () => {
+    // Arrange
+    const termsOfService = {
+      userHasAcceptedLatestTos: false,
+      permitsSystemUsage: true,
+    };
+
+    setupMockAjax(termsOfService);
+
+    // Act
+    await act(async () => {
+      render(h(TermsOfServicePage));
+    });
+
+    // Assert
+    screen.getByText('Continue under grace period');
+  });
+
+  it('does not show "Continue under grace period" when the user has not accepted the latest ToS and is not allowed to use Terra', async () => {
+    // Arrange
+    const termsOfService = {
+      userHasAcceptedLatestTos: false,
+      permitsSystemUsage: false,
+    };
+    setupMockAjax(termsOfService);
+
+    // Act
+    await act(async () => {
+      render(h(TermsOfServicePage));
+    });
+
+    // Assert
+    const continueUnderGracePeriodButton = screen.queryByText('Continue under grace period');
+    expect(continueUnderGracePeriodButton).not.toBeInTheDocument();
```

Comment:
Alternatively,

```suggestion
    expect(continueUnderGracePeriodButton).toBeNull();
```

A little shorter to type.